etcdserver/raft.go

```diff
@@ -230,10 +230,21 @@ func (r *raftNode) start(rh *raftReadyHandler) {
 
 				r.Advance()
 
-				if isCandidate {
+				if !islead && !waitApply {
```

Comment:
to make it safer, we probably should move the waiting before we send out the message to other peers at line 227.